tests/features/steps/artifacts-poetry-plugin-steps.py

```diff
@@ -15,52 +13,107 @@
 
 
 class Package:
-    def __init__(self, name, pretty_version) -> None:
+    def __init__(self, name, pretty_version, files) -> None:
         self.name = name
         self.pretty_version = pretty_version
+        self.files = files
 
     def name(self):
         return self.name
 
     def pretty_version(self):
         return self.pretty_version
 
+    def files(self):
+        return self.files
+
 
 @given("the following artifacts")
 def the_following_artifacts(context):
     context.packages = {}
     for row in context.table:
+        name = row["package"]
+        version = row["version"]
+        filename = f"{name}-{version}.whl"
+        file = open(os.path.join(TEST_RESOURCES_DIR, filename), "w")
+        file.close()
+        files = []
+        files.append(filename)
         context.packages[row["key"]] = Package(
-            name=row["package"], pretty_version=row["version"]
+            name=name, pretty_version=version, files=files
         )
 
 
-@given("a Python project with dependencies on artifacts {keys}")
+@given("a Python project with dependencies on package {keys}")
 def step_impl(context, keys):
-    shutil.copy(TEMPLATE_PY_PROJECT, PY_PROJECT)
-    with open(PY_PROJECT, "a") as py_project:
-        py_project.write("\n[tool.poetry.dependencies]\n")
+    with open(POETRY_LOCK, "w") as poetry_lock:
```

Comment:
Q: can we not have poetry generate the lock file so we don't have to modify this implementation as they change the internal details of the lock file?